From f5373d5a00b3e228955c3f136100dc3a0962c887 Mon Sep 17 00:00:00 2001
From: Patrick Rudolph <patrick.rudolph@9elements.com>
Date: Fri, 13 Oct 2023 08:34:49 +0200
Subject: [PATCH 13/13] configurations: Add Intel P4510 SSD

Add support for P4510 SSD by adding a new config.

Tested: Can read the temperature of P4510 SSD.

Change-Id: I2611af3ef65549ace7f6a066b7caea842e58d5b6
Signed-off-by: Patrick Rudolph <patrick.rudolph@9elements.com>
---
 configurations/nvme_p4510.json | 54 ++++++++++++++++++++++++++++++++++
 meson.build                    |  1 +
 2 files changed, 55 insertions(+)
 create mode 100644 configurations/nvme_p4510.json

diff --git a/configurations/nvme_p4510.json b/configurations/nvme_p4510.json
new file mode 100644
index 0000000..01622e3
--- /dev/null
+++ b/configurations/nvme_p4510.json
@@ -0,0 +1,54 @@
+{
+    "Bus": "$bus",
+    "Exposes": [
+        {
+            "Address": "$address",
+            "Bus": "$bus",
+            "Name": "NVMe $index FRU",
+            "Type": "EEPROM"
+        },
+        {
+            "Address": "0x6a",
+            "Bus": "$bus",
+            "Name": "NVMe $index Temp",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 115
+                },
+                {
+                    "Direction": "greater than",
+                    "Name": "upper non critical",
+                    "Severity": 0,
+                    "Value": 110
+                },
+                {
+                    "Direction": "less than",
+                    "Name": "lower non critical",
+                    "Severity": 0,
+                    "Value": 5
+                },
+                {
+                    "Direction": "less than",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 0
+                }
+            ],
+            "Type": "NVME1000"
+        }
+    ],
+    "Logging": "Off",
+    "Name": "NVMe $index",
+    "Probe": "xyz.openbmc_project.FruDevice({'PRODUCT_PRODUCT_NAME': 'P4510\\w?'})",
+    "Type": "NVMe",
+    "xyz.openbmc_project.Inventory.Decorator.Asset": {
+        "Manufacturer": "$PRODUCT_MANUFACTURER",
+        "Model": "$PRODUCT_PRODUCT_NAME",
+        "PartNumber": "$PRODUCT_PART_NUMBER",
+        "SerialNumber": "$PRODUCT_SERIAL_NUMBER"
+    },
+    "xyz.openbmc_project.Inventory.Item.Chassis": {}
+}
diff --git a/meson.build b/meson.build
index 42d00c0..6f124bd 100644
--- a/meson.build
+++ b/meson.build
@@ -130,6 +130,7 @@ configs = [
     'mudflap.json',
     'nisqually.json',
     'nvme_p4000.json',
+    'nvme_p4510.json',
     'pcie_ssd_retimer.json',
     'pennybacker.json',
     'pssf132202a.json',
-- 
2.41.0

